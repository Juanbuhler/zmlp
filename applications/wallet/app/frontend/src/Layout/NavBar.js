import ProjectSwitcher from '../ProjectSwitcher'
import { colors, spacing } from '../Styles'
import LogoSvg from './logo.svg'

const LOGO_WIDTH = 110

const LayoutNavBar = () => {
  return (
    <div
      css={{
        display: 'flex',
        justifyContent: 'left',
        alignItems: 'center',
        backgroundColor: colors.grey1,
        padding: spacing.small,
      }}>
      <LogoSvg width={LOGO_WIDTH} />
      <ProjectSwitcher />
    </div>
  )
}

export default LayoutNavBar
