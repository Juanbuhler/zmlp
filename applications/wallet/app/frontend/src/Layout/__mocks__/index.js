import { createElement } from 'react'

const Layout = ({ children, ...rest }) =>
  createElement('Layout', rest, children(rest))

export default Layout
