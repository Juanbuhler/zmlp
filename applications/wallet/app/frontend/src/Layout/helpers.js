export const getJobId = ({ url }) => {
  const projects = url.indexOf('projects')
  return url.substr(projects + 9).split('/')[0]
}
