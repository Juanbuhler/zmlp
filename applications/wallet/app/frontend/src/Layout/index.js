import { useState, useEffect, useRef } from 'react'
import PropTypes from 'prop-types'
import {
  disableBodyScroll,
  enableBodyScroll,
  clearAllBodyScrollLocks,
} from 'body-scroll-lock'

import userShape from '../User/shape'

import { constants } from '../Styles'

import Navbar from '../Navbar'
import Sidebar from '../Sidebar'

const Layout = ({ user, results, logout, children }) => {
  const sidebarRef = useRef()

  const [isSidebarOpen, setSidebarOpen] = useState(false)

  const [selectedProject, setSelectedProject] = useState({
    id: results[0].id,
    name: results[0].name,
  })

  const projects = results.map(({ id, name }) => {
    return {
      id,
      name,
      selected: selectedProject.id === id,
    }
  })

  useEffect(() => {
    if (isSidebarOpen) disableBodyScroll(sidebarRef.current)
    if (!isSidebarOpen) enableBodyScroll(sidebarRef.current)
    return clearAllBodyScrollLocks
  }, [isSidebarOpen, sidebarRef])

  return (
    <div css={{ height: '100%' }}>
      <Navbar
        user={user}
        isSidebarOpen={isSidebarOpen}
        projects={projects}
        setSidebarOpen={setSidebarOpen}
        setSelectedProject={setSelectedProject}
        logout={logout}
      />
      <Sidebar
        isSidebarOpen={isSidebarOpen}
        setSidebarOpen={setSidebarOpen}
        ref={sidebarRef}
      />
      <div css={{ paddingTop: constants.navbar.height }}>
        {children({ selectedProject })}
      </div>
    </div>
  )
}

Layout.propTypes = {
  results: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string.isRequired,
      name: PropTypes.string.isRequired,
    }),
  ).isRequired,
  user: PropTypes.shape(userShape).isRequired,
  logout: PropTypes.func.isRequired,
  children: PropTypes.func.isRequired,
}

export default Layout
