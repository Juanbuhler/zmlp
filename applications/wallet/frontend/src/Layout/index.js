import { useState, useEffect, useRef } from 'react'
import PropTypes from 'prop-types'
import {
  disableBodyScroll,
  enableBodyScroll,
  clearAllBodyScrollLocks,
} from 'body-scroll-lock'

import userShape from '../User/shape'

import { constants, spacing } from '../Styles'

import Navbar from '../Navbar'
import Sidebar from '../Sidebar'

const Layout = ({ user, logout, children }) => {
  const sidebarRef = useRef()

  const [isSidebarOpen, setSidebarOpen] = useState(false)

  useEffect(() => {
    if (isSidebarOpen) disableBodyScroll(sidebarRef.current)
    if (!isSidebarOpen) enableBodyScroll(sidebarRef.current)
    return clearAllBodyScrollLocks
  }, [isSidebarOpen, sidebarRef])

  return (
    <div css={{ height: '100%' }}>
      <Navbar
        user={user}
        isSidebarOpen={isSidebarOpen}
        setSidebarOpen={setSidebarOpen}
        logout={logout}
      />
      <Sidebar
        isSidebarOpen={isSidebarOpen}
        setSidebarOpen={setSidebarOpen}
        ref={sidebarRef}
      />
      <div
        css={{
          marginTop: constants.navbar.height,
          paddingLeft: spacing.spacious,
          paddingRight: spacing.spacious,
          paddingBottom: spacing.spacious,
          height: `calc(100vh - ${constants.navbar.height}px)`,
          display: 'flex',
          flexDirection: 'column',
        }}>
        {children}
      </div>
    </div>
  )
}

Layout.propTypes = {
  user: PropTypes.shape(userShape).isRequired,
  logout: PropTypes.func.isRequired,
  children: PropTypes.node.isRequired,
}

export default Layout
