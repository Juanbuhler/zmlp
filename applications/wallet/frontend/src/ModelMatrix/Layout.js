import { useReducer } from 'react'

// TODO: fetch data
import matrix from './__mocks__/matrix'

import { colors, constants, spacing, typography } from '../Styles'

import Button, { VARIANTS } from '../Button'

import PreviewSvg from '../Icons/preview.svg'

import { INITIAL_STATE, reducer } from './reducer'

import ModelMatrixControls from './Controls'
import ModelMatrixMatrix from './Matrix'
import ModelMatrixPreview from './Preview'

const PANEL_WIDTH = 200

const ModelMatrixLayout = () => {
  const [settings, dispatch] = useReducer(reducer, INITIAL_STATE)

  return (
    <div
      css={{
        width: '100%',
        height: '100%',
        display: 'flex',
        flexDirection: 'column',
      }}
    >
      <div
        css={{
          display: 'flex',
          alignItems: 'center',
          padding: spacing.normal,
          borderBottom: constants.borders.regular.coal,
          fontSize: typography.size.medium,
          lineHeight: typography.height.medium,
          backgroundColor: colors.structure.lead,
        }}
      >
        <span
          css={{
            fontWeight: typography.weight.bold,
            paddingRight: spacing.small,
          }}
        >
          Overall Accuracy:
        </span>
        98%
        <ModelMatrixControls
          matrix={matrix}
          settings={settings}
          dispatch={dispatch}
        />
      </div>

      <div
        css={{
          flex: 1,
          width: '100%',
          display: 'flex',
          backgroundColor: colors.structure.lead,
        }}
      >
        <ModelMatrixMatrix
          matrix={matrix}
          settings={settings}
          dispatch={dispatch}
        />

        {settings.isPreviewOpen && (
          <div
            css={{
              display: 'flex',
              flexDirection: 'column',
              width: PANEL_WIDTH,
              height: '100%',
              borderLeft: constants.borders.regular.coal,
              overflow: 'auto',
            }}
          >
            <ModelMatrixPreview
              selectedCell={settings.selectedCell}
              matrix={matrix}
            />
          </div>
        )}

        <div
          css={{
            display: 'flex',
            flexShrink: 0,
            alignItems: 'flex-start',
            borderLeft: constants.borders.regular.coal,
          }}
        >
          <Button
            aria-label="Preview"
            title="Preview"
            variant={VARIANTS.ICON}
            onClick={() =>
              dispatch({
                isPreviewOpen: !settings.isPreviewOpen,
              })
            }
            style={{
              flex: 'none',
              paddingTop: spacing.normal,
              paddingBottom: spacing.normal,
              borderBottom: constants.borders.regular.coal,
              color: settings.isPreviewOpen
                ? colors.key.one
                : colors.structure.steel,
              ':hover': {
                backgroundColor: colors.structure.mattGrey,
              },
              borderRadius: constants.borderRadius.none,
            }}
          >
            <PreviewSvg width={constants.icons.regular} />
          </Button>
        </div>
      </div>
    </div>
  )
}

export default ModelMatrixLayout
