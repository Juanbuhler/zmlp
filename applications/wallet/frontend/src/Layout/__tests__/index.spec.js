import TestRenderer, { act } from 'react-test-renderer'

import mockUser from '../../User/__mocks__/user'

import Layout from '..'

jest.mock('../../ProjectSwitcher', () => 'ProjectSwitcher')
jest.mock('../../Sidebar', () => 'Sidebar')

const noop = () => () => {}

describe('<Layout />', () => {
  it('should render properly', () => {
    const component = TestRenderer.create(
      <Layout user={mockUser} logout={noop}>
        Hello World
      </Layout>,
    )

    expect(component.toJSON()).toMatchSnapshot()

    act(() => {
      component.root
        .findByProps({ 'aria-label': 'Open Sidebar Menu' })
        .props.onClick({ preventDefault: noop })
    })

    expect(component.toJSON()).toMatchSnapshot()
  })
})
