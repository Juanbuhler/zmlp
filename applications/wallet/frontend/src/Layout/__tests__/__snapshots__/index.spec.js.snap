// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> should render properly 1`] = `
.emotion-11 {
  height: 100%;
}

.emotion-9 {
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  height: 44px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #2e2e2e;
  box-shadow: 0 0 4px 0 rgba(0,0,0,0.5);
  z-index: 100;
  padding-left: 16px;
  padding-right: 16px;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
}

.emotion-0 {
  border: 0;
  background-color: inherit;
  color: #808080;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 8px;
  margin: 0;
  margin-left: -8px;
  cursor: pointer;
}

.emotion-1 {
  padding-left: 8px;
  padding-right: 8px;
}

.emotion-8 {
  margin-right: -12px;
}

.emotion-7 {
  position: relative;
}

.emotion-6 {
  -webkit-flex: 0;
  -ms-flex: 0;
  flex: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 2px;
  margin: 0;
  padding: 12px 32px;
  white-space: nowrap;
  font-size: 15px;
  line-height: 18px;
  font-weight: 500;
  border: none;
  cursor: pointer;
  color: #ffffff;
  background-color: transparent;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  padding: 4px 8px;
  color: #73b61c;
  font-weight: 400;
  height: 100%;
}

.emotion-6:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-6[aria-disabled=true] {
  color: #2e2e2e;
}

.emotion-6:hover {
  background-color: #3b3b3b;
}

.emotion-6:hover {
  cursor: pointer;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: 0;
  margin: 0;
  padding: 0;
  width: 28px;
  height: 28px;
  border-radius: 28px;
  color: #252525;
  background-color: #808080;
  font-weight: 700;
}

.emotion-4 {
  margin-left: 8px;
}

.emotion-10 {
  margin-top: 44px;
  padding-left: 32px;
  padding-right: 32px;
  padding-bottom: 32px;
  height: calc(100vh - 44px);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

<div
  className="emotion-11"
>
  <div
    className="emotion-9"
  >
    <div
      className="emotion-2"
    >
      <button
        aria-label="Open Sidebar Menu"
        className="emotion-0"
        onClick={[Function]}
        type="button"
      >
        <svg
          aria-hidden={true}
          data-file-name="hamburger.svg"
          width={20}
        />
      </button>
      <div
        className="emotion-1"
      >
        <svg
          data-file-name="logo.svg"
          width={110}
        />
      </div>
      <ProjectSwitcher />
    </div>
    <div
      className="emotion-8"
    >
      <div
        className="emotion-7"
      >
        <button
          aria-label="Open user menu"
          className="emotion-6"
          onBlur={[Function]}
          onClick={[Function]}
          type="button"
        >
          <div
            className="emotion-5"
          >
            <div
              className="emotion-3"
            >
              JD
            </div>
            <svg
              className="emotion-4"
              color="#808080"
              data-file-name="chevron.svg"
              width={20}
            />
          </div>
        </button>
      </div>
    </div>
  </div>
  <Sidebar
    isSidebarOpen={false}
    setSidebarOpen={[Function]}
  />
  <div
    className="emotion-10"
  >
    Hello World
  </div>
</div>
`;

exports[`<Layout /> should render properly 2`] = `
.emotion-11 {
  height: 100%;
}

.emotion-9 {
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  height: 44px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #2e2e2e;
  box-shadow: 0 0 4px 0 rgba(0,0,0,0.5);
  z-index: 100;
  padding-left: 16px;
  padding-right: 16px;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
}

.emotion-0 {
  border: 0;
  background-color: inherit;
  color: #808080;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 8px;
  margin: 0;
  margin-left: -8px;
  cursor: pointer;
}

.emotion-1 {
  padding-left: 8px;
  padding-right: 8px;
}

.emotion-8 {
  margin-right: -12px;
}

.emotion-7 {
  position: relative;
}

.emotion-6 {
  -webkit-flex: 0;
  -ms-flex: 0;
  flex: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 2px;
  margin: 0;
  padding: 12px 32px;
  white-space: nowrap;
  font-size: 15px;
  line-height: 18px;
  font-weight: 500;
  border: none;
  cursor: pointer;
  color: #ffffff;
  background-color: transparent;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  padding: 4px 8px;
  color: #73b61c;
  font-weight: 400;
  height: 100%;
}

.emotion-6:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-6[aria-disabled=true] {
  color: #2e2e2e;
}

.emotion-6:hover {
  background-color: #3b3b3b;
}

.emotion-6:hover {
  cursor: pointer;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: 0;
  margin: 0;
  padding: 0;
  width: 28px;
  height: 28px;
  border-radius: 28px;
  color: #252525;
  background-color: #808080;
  font-weight: 700;
}

.emotion-4 {
  margin-left: 8px;
}

.emotion-10 {
  margin-top: 44px;
  padding-left: 32px;
  padding-right: 32px;
  padding-bottom: 32px;
  height: calc(100vh - 44px);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

<div
  className="emotion-11"
>
  <div
    className="emotion-9"
  >
    <div
      className="emotion-2"
    >
      <button
        aria-label="Open Sidebar Menu"
        className="emotion-0"
        onClick={[Function]}
        type="button"
      >
        <svg
          aria-hidden={true}
          data-file-name="hamburger.svg"
          width={20}
        />
      </button>
      <div
        className="emotion-1"
      >
        <svg
          data-file-name="logo.svg"
          width={110}
        />
      </div>
      <ProjectSwitcher />
    </div>
    <div
      className="emotion-8"
    >
      <div
        className="emotion-7"
      >
        <button
          aria-label="Open user menu"
          className="emotion-6"
          onBlur={[Function]}
          onClick={[Function]}
          type="button"
        >
          <div
            className="emotion-5"
          >
            <div
              className="emotion-3"
            >
              JD
            </div>
            <svg
              className="emotion-4"
              color="#808080"
              data-file-name="chevron.svg"
              width={20}
            />
          </div>
        </button>
      </div>
    </div>
  </div>
  <Sidebar
    isSidebarOpen={true}
    setSidebarOpen={[Function]}
  />
  <div
    className="emotion-10"
  >
    Hello World
  </div>
</div>
`;
